synapse/server/nodes/spike_detect.py

```diff
@@ -59,37 +60,46 @@ async def run(self):
                 self.logger.warning("Received non-broadband data")
                 continue
 
+            if self.buffer_start_ts is None:
+                self.buffer_start_ts = data.t0
+
             for channel_id, samples in data.samples:
                 self.channel_buffers[channel_id].extend(samples)
 
             refractory_period_in_samples = int(REFRACTORY_PERIOD_S * data.sample_rate)
             bin_size_in_samples = int(self.bin_size_ms * data.sample_rate / 1000)
 
-            while any(
+            while all(
```

Comment:
There's also a change here worth noting: previously we cut spike bins as soon as _any_ channel buffer reached the desired bin size, which could lead to an edge case where a channel that's missing a sample for whatever reason could fall behind by a whole bin, and never catch up.

Now, we are waiting until _all_ channels are full. 

Again, this is a fairly extreme edge case, so should not lead to increased latency.

Reply:
there should be some backstop timeout here so it never just hangs in case something strange happens